src/UI/StructureInspector.cpp

```diff
@@ -91,28 +91,28 @@ void StructureInspector::btnCloseClicked()
 
 void StructureInspector::drawPopulationRequirements()
 {
-	Renderer& r = Utility<Renderer>::get();
-	float posX = rect().x() + 10.0f;
-	float posY = rect().y() + 85.0f;
-
-	r.drawText(*FONT_BOLD, "Population Required", rect().x() + 10, posY, 255, 255, 255);
-
-	posY += 20;
-
-	if (mStructure->populationRequirements()[0] > 0)
-	{
-		std::string format = string_format("Workers: %i/%i", mStructure->populationAvailable()[0], mStructure->populationRequirements()[0]);
-		Color color = mStructure->populationAvailable()[0] >= mStructure->populationRequirements()[0] ? Color::White : Color::Red;
-		r.drawText(*FONT, format, posX, posY, color.red(), color.green(), color.blue());
-		posY += 10;
+	Renderer& renderer = Utility<Renderer>::get();
+
+	auto position = rect().startPoint() + NAS2D::Vector{10, 85};
+	renderer.drawText(*FONT_BOLD, "Population Required", position, NAS2D::Color::White);
+
+	const std::array<std::string, 2> populationTypes = {{
+		"Workers",
+		"Scientists"
+	}};
+
+	position.y() += 20;
+	for (std::size_t populationType = 0; populationType < populationTypes.size(); ++populationType) {
```

Comment:
Consider `for(auto populationType = 0u; ...` or `for(auto populationType = std::size_t{0u} ...` if it gets deduced to the wrong type.

Reply:
Using `auto` with `0u` would deduce the wrong type. It would give `unsigned int` rather than `std::size_t`. Depending on the platform, those types may be different.

I prefer putting the type on the left in the particular case. I don't see any clear benefit to the `= std::size_t{0u}` syntax. Seems to just add syntactic noise.

Reply:
I was in the "almost always auto" mindset. This mostly applies to generic code or situations where the type is repeated. I'm okay with leaving it as `std::size_t populationType = 0`.

Reply:
Yeah. Counting `for` loops seems to be one of the areas where `auto` hasn't managed to penetrate well. I think there were supposed to be some improvements coming to make counting `for` loops easier to write, and play nicer with `auto`. I believe the change involved Ranges.

Reply:
...and iterators: `for(auto iter = std::begin(vecVar) ... `